keadm/cmd/keadm/app/cmd/util/helminstaller.go

```diff
@@ -0,0 +1,498 @@
+package util
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+	"io/fs"
+	"io/ioutil"
+	"os"
+	"strings"
+	"time"
+
+	"helm.sh/helm/v3/pkg/action"
+	"helm.sh/helm/v3/pkg/release"
+	"helm.sh/helm/v3/pkg/storage/driver"
+	"helm.sh/helm/v3/pkg/strvals"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"sigs.k8s.io/yaml"
+
+	keCharts "github.com/kubeedge/kubeedge/build/helm/charts"
+	"github.com/kubeedge/kubeedge/common/constants"
+	types "github.com/kubeedge/kubeedge/keadm/cmd/keadm/app/cmd/common"
+)
+
+const (
+	CloudCoreHelmComponent = "cloudcore"
+	CloudCoreSubDir        = "cloudcore"
+
+	DefaultBaseHelmDir   = ""
+	DefaultAddonsHelmDir = "addons"
+	DefaultProfilesDir   = "profiles"
+	// DefaultProfileFilename is the name of the default profile yaml file.
+	DefaultProfileFilename = "version.yaml"
+	DefaultHelmValuesPath  = "values.yaml"
+
+	DefaultHelmTimeout = time.Duration(60 * time.Second)
+
+	DefaultHelmInstall  = true
+	DefaultHelmWait     = true
+	DefaultHelmCreateNs = true
+)
+
+var (
+	ErrListProfiles = errors.New("can not list profiles")
+)
+
+// KubeCloudHelmInstTool embeds Common struct
+// It implements ToolsInstaller interface
+type KubeCloudHelmInstTool struct {
+	Common
+	AdvertiseAddress string
+	Manifests        string
+	Namespace        string
+	CloudcoreImage   string
+	CloudcoreTag     string
+	IptablesMgrImage string
+	IptablesMgrTag   string
+	Sets             []string
+	Profile          string
+	ProfileKey       string
+	ExternalHelmRoot string
+	Force            bool
+	SkipCRDs         bool
+	DryRun           bool
+	Action           string
+	existsProfile    bool
+}
+
+// InstallTools downloads KubeEdge for the specified version
+// and makes the required configuration changes and initiates cloudcore.
+func (cu *KubeCloudHelmInstTool) InstallTools() error {
+	cu.SetOSInterface(GetOSInterface())
+	cu.SetKubeEdgeVersion(cu.ToolVersion)
+
+	baseHelmRoot := DefaultBaseHelmDir
+	if cu.ExternalHelmRoot != "" {
+		baseHelmRoot = cu.ExternalHelmRoot
+	}
+
+	switch cu.Action {
+	case types.HelmInstallAction:
+		if err := cu.RunHelmInstall(baseHelmRoot); err != nil {
+			return err
+		}
+	case types.HelmManifestAction:
+		if err := cu.RunHelmManifest(baseHelmRoot); err != nil {
+			return err
+		}
+	default:
+		fmt.Println("Not support this action")
+	}
+
+	return nil
+}
+
+// RunHelmInstall renders the Charts with the given values, then installs the Charts to the cluster.
+func (cu *KubeCloudHelmInstTool) RunHelmInstall(baseHelmRoot string) error {
+	// --force would not care about whether the cloud components exist or not
+	// Also, if gives a external helm root, no need to check and verify. Because it is always not a cloudcore.
+	if !cu.Force && cu.ExternalHelmRoot == "" {
+		cloudCoreRunning, err := cu.IsKubeEdgeProcessRunning(KubeCloudBinaryName)
+		if err != nil {
```

Comment:
This function seems to use `pid cloudcore 2>&1`  command to predict whether cloudcore is running or not. For helm install mode, we couldn't use this way. we could check pods to do it? @zhu733756

Reply:
> his function seems to use `pid cloudcore 2>&1` command to predict whether cloudcore is running or not. For helm install mode, we couldn't use this way. we could check pods to do it? @zhu733756

Yeah, checking pods is the correcting style of the k8s, last time I see this function can cover it, so reuse it.  I wonder if the former deployment mode is binary mode, we can also use it to check.